lib/utils/compose_ts.ts

```diff
@@ -313,14 +313,23 @@ function setTaskAttributes({
 		// multibuild (splitBuildStream) parses the composition internally so
 		// any tags we've set before are lost; re-assign them here
 		task.tag ??= [projectName, task.serviceName].join('_').toLowerCase();
-
-		task.dockerOpts ??= {};
 		if (isBuildConfig(d.image)) {
 			d.image.tag = task.tag;
-			if (d.image.args) {
-				task.dockerOpts.buildargs ??= {};
-				_.merge(task.dockerOpts.buildargs, d.image.args);
-			}
+		}
+		// reassign task.args so that the `--buildArg` flag takes precedence
+		// over assignments in the docker-compose.yml file (service.build.args)
+		task.args = {
+			...task.args,
+			...buildOpts.buildargs,
```

Comment:
This fixes issue #1053. Previously, `d.image.args` overwrote the `--buildArg` command-line option(s). Now `--buildArg` takes precedence, in line with the behavior of the standard `docker-compose` command-line tool.